pkg/infrastructure/aws/clusterapi/aws.go

```diff
@@ -254,48 +263,56 @@ func removeSSHRule(ctx context.Context, cl k8sClient.Client, infraID string) err
 	}
 	logrus.Debug("Updated AWSCluster to remove bootstrap SSH rule")
 
+	var sgID string
+	if sg, ok := awsCluster.Status.Network.SecurityGroups[capa.SecurityGroupControlPlane]; ok && len(sg.ID) > 0 {
+		sgID = sg.ID
+	} else if ok {
+		return fmt.Errorf("control plane security group id is not populated in awscluster status")
+	} else {
+		keys := make([]capa.SecurityGroupRole, 0, len(awsCluster.Status.Network.SecurityGroups))
+		for sgr := range awsCluster.Status.Network.SecurityGroups {
+			keys = append(keys, sgr)
+		}
+		return fmt.Errorf("controlplane not found in cluster security groups: %v", keys)
+	}
+
 	timeout := 15 * time.Minute
-	untilTime := time.Now().Add(timeout)
-	warnTime := time.Now().Add(5 * time.Minute)
-	warned := false
+	startTime := time.Now()
+	untilTime := startTime.Add(timeout)
 	timezone, _ := untilTime.Zone()
-	logrus.Infof("Waiting up to %v (until %v %s) for bootstrap SSH rule to be destroyed...", timeout, untilTime.Format(time.Kitchen), timezone)
-	if err := wait.ExponentialBackoffWithContext(ctx, wait.Backoff{
-		Duration: time.Second * 10,
-		Factor:   float64(1.5),
-		Steps:    32,
-		Cap:      timeout,
-	}, func(ctx context.Context) (bool, error) {
-		c := &capa.AWSCluster{}
-		if err := cl.Get(ctx, key, c); err != nil {
-			return false, err
-		}
-		if time.Now().After(warnTime) && !warned {
-			logrus.Warn("Deleting bootstrap SSH rule is still progressing but taking longer than expected")
-			warned = true
-		}
-		if sg, ok := c.Status.Network.SecurityGroups[capa.SecurityGroupControlPlane]; ok {
-			for _, r := range sg.IngressRules {
-				if r.Description == awsmanifest.BootstrapSSHDescription {
-					logrus.Debugf("Still waiting for bootstrap SSH security rule %s to be deleted from %s...", r.Description, sg.ID)
-					return false, nil
+	logrus.Debugf("Waiting up to %v (until %v %s) for bootstrap SSH rule to be destroyed...", timeout, untilTime.Format(time.Kitchen), timezone)
+	if err := wait.PollUntilContextTimeout(ctx, 15*time.Second, timeout, true,
+		func(ctx context.Context) (bool, error) {
+			sgs, err := awsconfig.DescribeSecurityGroups(ctx, session, []string{sgID}, region)
+			if err != nil {
+				return false, fmt.Errorf("error getting security group: %w", err)
+			}
+			noPublicSSHRulesFound := true
```

Comment:
Nit: maybe we can consider inverting this? I think it'd be easier to read:
```suggestion
			publicSSHRulesFound := false
```